fastapi/readme/USAGE.rst

```diff
@@ -0,0 +1,1185 @@
+endpoint, you can define the app that you want to mount in the **'app'** field
+and the path where you want to mount it in the **'path'** field.
+
+figure:: static/description/endpoint.png
+
+    FastAPI Endpoint
+
+Thanks to the **'fastapi.endpoint'** model, you can create as many endpoints as
+you wand and mount as many apps as you want in each endpoint. The endpoint is
+also the place where you can define configuration parameters for your app. A
+typical example is the authentication method that you want to use for your app
+when accessed at the endpoint path.
+
+Now, you can create your first router. For that, you need to define a global
+variable into your fastapi_endpoint module called for example 'demo_api_router'
+
+.. code-block:: python
+
+    from fastapi import APIRouter
+    from odoo import fields, models
+
+    class FastapiEndpoint(models.Model):
+
+        _inherit = "fastapi.endpoint"
+
+        app: str = fields.Selection(
+            selection_add=[("demo", "Demo Endpoint")], ondelete={"demo": "cascade"}
+        )
+
+    # create a router
+    demo_api_router = APIRouter()
+
+
+To make your router available to your app, you need to add it to the list of routers
+returned by the **get_fastapi_routers** method of your fastapi_endpoint model.
+
+.. code-block:: python
+
+    from fastapi import APIRouter
+    from odoo import api, fields, models
+
+    class FastapiEndpoint(models.Model):
+
+        _inherit = "fastapi.endpoint"
+
+        app: str = fields.Selection(
+            selection_add=[("demo", "Demo Endpoint")], ondelete={"demo": "cascade"}
+        )
+
+        @api.model
+        def get_fastapi_routers(self):
+            if self.app == "demo":
+                return [demo_api_router]
+            return super().get_fastapi_routers()
+
+    # create a router
+    demo_api_router = APIRouter()
+
+Now, you can start adding routes to your router. For example, let's add a route
+that returns a list of partners.
+
+.. code-block:: python
+
+    from fastapi import APIRouter
+    from pydantic import BaseModel
+    from odoo import api, fields, models
+    from ..depends import odoo_env
+
+    class FastapiEndpoint(models.Model):
+
+        _inherit = "fastapi.endpoint"
+
+        app: str = fields.Selection(
+            selection_add=[("demo", "Demo Endpoint")], ondelete={"demo": "cascade"}
+        )
+
+        @api.model
+        def get_fastapi_routers(self):
+            if self.app == "demo":
+                return [demo_api_router]
+            return super().get_fastapi_routers()
+
+    # create a router
+    demo_api_router = APIRouter()
+
+    class PartnerInfo(BaseModel):
+        name: str
+        email: str
+
+    @demo_api_router.get("/partners", response_model=list[PartnerInfo])
+    def get_partners(env=Depends(odoo_env)) -> list[PartnerInfo]:
+        return [
+            PartnerInfo(name=partner.name, email=partner.email)
+            for partner in env["res.partner"].search([])
```

Comment:
This method cannot be tested. When running `search` the system crashes. The endpoint cannot access the database during the test. However, it can be executed from `Postman` e.g.